release/Magarena/scripts/Rowdy_Crew.groovy

```diff
@@ -0,0 +1,29 @@
+[
+    new EntersBattlefieldTrigger() {
+        @Override
+        public MagicEvent executeTrigger(final MagicGame game,final MagicPermanent permanent, final MagicPayedCost payedCost) {
+            return new MagicEvent(
+                permanent,
+                this,
+                "PN draws three cards, then PN discards two cards at random. " +
+                "If two cards that share a card type are discarded this way, put two +1/+1 counters on SN."
+            );
+        }
+        @Override
+        public void executeEvent(final MagicGame game, final MagicEvent event) {
+            game.doAction(new DrawAction(event.getPlayer(), 3));
+            game.addEvent(MagicDiscardEvent.Random(
```

Comment:
In this case, since we want to know the cards, we can't use the event as we cannot retrieve the cards. Checking the graveyard is no good, since there may be an effect that prevents them from going to the graveyard. The only way to do this is to decide on the cards in the effect and then use the DiscardCardAction to discard them.

Something like this

```
  final MagicCardList hand = new MagicCardList(event.getPlayer().getHand());
  final MagicCardList toDiscard = hand.getRandomCards(2);
  // discard the two cards
  // check if they share the same type...
```

Reply:
Another thing to note with events is that they don't happen right away, so we can't check their results immediately. addEvent puts the event in an event queue to await execution.